### DISCUSSION TITLE
My fork with 60 minute plot and automatic AP mode

### DISCUSSION THREAD

**Discussion by ehorvat1** [UtilitechAS/amsreader-firmware#567] [Show and tell]
Hi there, here my fork from Rev 2.2.17 which includes a 60 minute plot and some automatic switchover to AP Wifi mode (instead of reboot option). 
For details see: https://github.com/ehorvat1/AmsToMqttBridge-EHo . 
Please Note: The ESP8266 Flash became quite full (compiler told 99.9%) which lead to inability for OTA firmware updates. So I deleted the automatic firmware updates. 
Sorry for inserting comments into the code....search for EHorvat to see my changes....

BR, Engelbert

**Comment by ArnieO** [comment]
Thank you for your contribution! 👍 
I think OTA update for 8266 is a priority, so when added features cause capacity issues with 8266, it should be built in such a way that the new feature is only available on ESP32.

**Reply by ehorvat1** [reply]
This was a kind of test of my abilities in programmin...and since I have a 8266 in operation I had to try it for 8266. 
What do you think about the automatic switchover to AP mode ? I did this because on one location the smart meter is in a closed cabinet and I have to ask for its key to open all the time when I mess around with it. Of course if one runs in a cycle of reboot loop this wont help... then just USB/FDTI connection helps. BR, Engelbert

**Reply by gskjold** [reply]
I see the benefits of the automatic AP mode on WiFi loss, but there is also a security concern with this feature, as it would expose meter data to potentially unwanted access. It could be a good configuration option to have though, so I'll take notes for future additions to configuration

**Reply by ehorvat1** [reply]
Yes the automatic AP mode could introduce a safety risk. So I did write a quite long text right next to the check mark where it can be enabled. It reads: "Start a Wifi AP on home Wifi connection problem. Note: The AP is secured. But one can find the password on my github readme file. So if the AP can be publicly accessed you should think about setting some security using General sections Security settings" . I hope this makes someone think about security a bit....but at the end the end-user has to decide....

**Comment by haldi4803** [comment]
i really like the 10 minute an 1hour Plot.
Especially if you want to check how much certain applications use if you plug in the hairdryer or vacum cleaner or raclette oven.
Might also be really cool if you have a solar panels on the roof and want to check how much they just lost because a cloud flew over.

Any chance of those comming to official release?

**Reply by gskjold** [reply]
v2.3 which is currently under development has a 60min plot
